terminal/plugins/org.eclipse.tm.terminal.test/src/org/eclipse/tm/internal/terminal/emulator/MockTerminalControlForText.java

```diff
@@ -32,8 +32,10 @@ public void setState(TerminalState state) {
 	}
 
 	@Override
-	public void setTerminalTitle(String title) {
-		allTitles.add(title);
+	public void setTerminalTitle(String title, String requestor) {
+		if (requestor.equals("ANSI")) {
```

Comment:
This constant should be shared so that future readers of the code can track down how these things connect.

Have you considered using an `enum` to handle the requestor?

Reply:
Agree. What do you think is the best place for this enum?

Reply:
1. The enum can go right next to the main declaration i.e. an inner enum in org.eclipse.tm.internal.terminal.emulator.VT100TerminalControl.setTerminalTitle(String, String)
2. be clear in the API if null is allowed for the requestor (you have it documented in the implementation class, but that needs to be in the interface because that is the API)
3. All the places that you updated to call the new API don't really seem necessary, e.g. you don't need to update the one calling with "RCM" at all with the evolved API.